auth/Program.cs

```diff
@@ -15,18 +15,72 @@
  */
 // [START all]
 
+using Google.Apis.Auth.OAuth2;
 using Google.Apis.Storage.v1.Data;
 using Google.Cloud.Storage.V1;
 using System;
+using System.IO;
 
 namespace GoogleCloudSamples
 {
     public class AuthSample
     {
         const string usage = @"Usage:AuthSample <bucket_name>";
+        string _projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+
+        // [START auth_cloud_implicit]
+        void ImplicitAuth()
+        {
+            // If you don't specify credentials when constructing the client, the
+            // client library will look for credentials in the environment.
+            var storage = StorageClient.Create();
+            // Make an authenticated API request.
+            var buckets = storage.ListBuckets(_projectId);
+            foreach (var bucket in buckets)
+            {
+                Console.WriteLine(bucket.Name);
+            }
+        }
+        // [END auth_cloud_implicit]
+
+        void ExplicitAuth(string jsonPath)
+        {
+            // Explicitly use service account credentials by specifying the private key
+            // file.
+            GoogleCredential credential = null;
+            using (var jsonStream = new FileStream(jsonPath, FileMode.Open,
+                FileAccess.Read, FileShare.Read))
+            {
+                credential = GoogleCredential.FromStream(jsonStream);
+            }
+            var storage = StorageClient.Create(credential);
+            // Make an authenticated API request.
+            var buckets = storage.ListBuckets(_projectId);
+            foreach (var bucket in buckets)
+            {
+                Console.WriteLine(bucket.Name);
+            }
+        }
+
+        void ExplicitAuthComputeEngine()
+        {
+            // Explicitly use Compute Engine credentials. These credentials are
+            // available on Compute Engine, App Engine Flexible, and Container Engine.
+            var computeCredential = new ComputeCredential();
+            // How to instantiate?  This gives me a compile time error.

```

Comment:
There is currently no way to do this.
The [constructor you're trying to use](https://github.com/google/google-api-dotnet-client/blob/master/Src/Support/Google.Apis.Auth/OAuth2/GoogleCredential.cs#L45) is `internal`; and there's no `From...` method that accepts an `ICredential` or `ComputeCredential`.

We *could* make this possible, but I'm not very keen to. In real use I doubt anyone is wanting to do this.